File: Solutions/0754/0754.py
# -*- coding:utf-8 -*-
# Author: washing
# DateTime: 2022/11/4 10:18
# File: 0754.py
# Desc: 


class Solution:
    def reachNumber(self, target: int) -> int:
        target = abs(target)
        total = 0
        i = 0
        while total < target or (total - target) % 2 != 0:
            i += 1
            total += i
        return i

File: Solutions/0754/0754.c
/***************************************************************************************************
 * Copyright: washing
 * FileName: 0754.c
 * Author: washing
 * Version: 1.0
 * Date: 2022-11-04 11:32:22
***************************************************************************************************/


int reachNumber(int target){
    target = target > 0 ? target : -target;
    int total = 0;
    int i = 0;
    while (total < target || (total - target) % 2 != 0) {
        i++;
        total += i;
    }
    return i;
}
